web-service/helpers.py

```diff
@@ -380,6 +385,18 @@ def StoreUrl(siteInfo, url, content, encoding):
 
 ################################################################################
 
+def FaviconUrl(url):
+    # Make sure we only fetch favicons we've already added to our database.
```

Comment:
Also.. is there any reason why this is important to constrain?

If someone wants to abuse the feature, all they have to do is call "resolve-scan" once for a given url.  Perhaps we need to rethink this approach..